lib/lightning/kafka_triggers/pipeline.ex

```diff
@@ -0,0 +1,212 @@
+defmodule Lightning.KafkaTriggers.Pipeline do
+  @moduledoc """
+  Broadway pipeline that processes messages from Kafka clusters and persists
+  the received messages if they are not duplicating a previous message.
+  """
+  use Broadway
+
+  alias Ecto.Multi
+  alias Lightning.KafkaTriggers
+  alias Lightning.KafkaTriggers.TriggerKafkaMessage
+  alias Lightning.KafkaTriggers.TriggerKafkaMessageRecord
+  alias Lightning.Repo
+  alias Lightning.Workflows.Trigger
+
+  require Logger
+
+  def start_link(opts) do
+    trigger_id = opts |> Keyword.get(:trigger_id)
+
+    Broadway.start_link(__MODULE__,
+      name: trigger_id,
+      context: %{
+        trigger_id: trigger_id
+      },
+      producer: [
+        module: {
+          BroadwayKafka.Producer,
+          build_producer_opts(opts)
+        },
+        concurrency: 1
+      ],
+      processors: [
+        default: [
+          concurrency: 10
+        ]
+      ],
+      batchers: []
+    )
+  end
+
+  @impl true
+  def handle_message(_processor, message, context) do
```

Comment:
It might have been a good approach to start with, but for 100 messages with `handle_batch` we can reduce the amount of inserts and updates from 300 to 3. Or if there is no limit for the message size, we can decrease the batch size to 10 and leverage the `Multi.insert_all`.